packages/pagination/test/helpers.ts

```diff
@@ -89,6 +96,22 @@ class FakeAppController {
   public async testPagination(@MongoPaginationParamDecorator({}) pagination: MongoPagination): Promise<{}> {
     return { pagination };
   }
+
+  /**
+   * Find all documents with the new query pagination
+   */
+  @UseInterceptors(new PaginationBodyInterceptor({ defaultLimit: 200, pageName: 'page', perPageName: 'limit' }))
+  @Get('/resource')
+  public async findAllWithQueryParam(): Promise<DataToPaginate<FakeDataToReturn>> {
+    const data: FakeDataToReturn[] = [];
+    const maxDocuments: number = 1015;
+
+    for (let i: number = 0; i < maxDocuments; i++) {
+      data.push({ name: `doc_${i}`, index: i, createdAt: new Date() });
+    }
+
+    return { totalResources: data.length, resources: data };
```

Comment:
Why not keeping the exact same structure as the link interceptor? With `totalDocs` and `resource` (a `s` is missing 😞 we could include a breaking change to update this)

Reply:
What about keeping totalDocs and resource ? @Bendakh  It would be easy and without effects to change it on our side when you will use the nestjs-component

Reply:
It makes more sense to have `resources` instead of `resource` since we expect an array. That's why we could include a breaking change in this Pull Request

Reply:
Ok. What about totalResources vs totalDocs ?

Reply:
As you wish, as long as it is the same for both of the interceptors 😄

Reply:
then totalResources is more coherent